src/e2e/specs/dashboard.spec.ts

```diff
@@ -302,6 +302,58 @@ test.describe(`${process.env.E2E_TEST_ENV} - Breaches Dashboard  - Payment`, ()
       `${process.env.E2E_TEST_ENV === "prod" ? "yearly" : "every ⁨2⁩ months"}`,
     );
   });
+
+  test("Verify that the Premium upsell screen is displayed correctly - overview card", async ({
+    dashboardPage,
+    // purchasePage,
+    page,
+  }) => {
+    test.info().annotations.push({
+      type: "testrail",
+      description:
+        "https://testrail.stage.mozaws.net/index.php?/cases/view/2463625",
+    });
+
+    //checking that the user can reach upsell page
+    await dashboardPage.goToDashboard();
+    const upsellButton = page.locator("section > div a");
+    await expect(upsellButton).toBeVisible();
+    await upsellButton.click();
+    await page.waitForURL(/.*\/fix\/.*\/view-data-brokers\/?/);
+    const removeThemButton = page.getByText("Remove them for me");
+    await removeThemButton.click();
+
+    //checking the bullet points
+    await page.waitForURL(/.*\/fix\/.*\/automatic-remove\/?/);
+    const ulElement = page.locator("div > strong + ul");
+    expect(await ulElement.isVisible()).toBeTruthy();
+
+    const expectedItems = [
+      "Monthly scan of ⁨190⁩ data broker sites that may be selling your personal info",
+      "Automatic data removal from sites that are selling your personal info",
+      "Guided experience through high risk data breaches that require manual steps",
+      "Continuous monitoring for new exposures",
+      "Alerts when your data has been breached",
+    ];
+
+    for (const itemText of expectedItems) {
+      const liElement = page.locator(`ul li:has-text("${itemText}")`).first();
```

Comment:
You can also nest locators. For example, you could do `const liElems = ulElement.locator("/li")` I believe to grab all the \<li> elements.

Then you can use things like liElems.count() and go through them one by one similarly to how you did here.

But in my opinion how you did it here achieves the same thing, so I don't see a problem with this.

Reply:
Seems like Locator does not support 'for..of', which makes the code more complicated than it should be. I think I'll put the expected contents inside the automaticRemovePage POM, which will hopefully make it cleaner